### DISCUSSION TITLE
How do we share the same CSS property across variants?

### DISCUSSION THREAD

**Discussion by seeden** [tamagui/tamagui#84] [Q&A]
Consider following example where two variants are using same CSS property:

```tsx
const Text = styled(SizableText, {
  variants: {
    underline: {
      true: {
        textDecoration: 'underline',
      },
      false: {
        textDecoration: 'none',
      },
    },
    strikethrough: {
      true: {
        textDecoration: 'line-through',
      },
      false: {
        textDecoration: 'none',
      },
    },
  },
});
```

Next render of Text component will use textDecoration === 'none' for media >= lg

```tsx
    <Text
      underline
      
      $md={{
        underline: false,
        strikethrough: true,
      }}
      $lg={{
        underline: true,
        strikethrough: false,
      }}
    >TeStBlABla</Text>
```

and following render of Text component will use textDecoration === 'underline' for media >= lg

```tsx
    <Text
      underline
      
      $md={{
        underline: false,
        strikethrough: true,
      }}
      $lg={{
        strikethrough: false,
        underline: true,
      }}
    >TeStBlABla</Text>
```

Just order was changed.
FYI: order of properties is not consistent across browsers and therefore, users can see different results.

Do you have a better idea how to work with same css properties across variants?

**Comment by natew** [comment]
This sounds like a bug, media queries should *always* win based on the order of the given media query in the `createTamagui` definition, but the *props within them* should always win based on their order at runtime definition. I'll give it a look.

**Comment by natew** [comment]
I did add some docs to prop ordering to the site as I think it's a huge and cool feature of Tamagui and not discussed before: https://github.com/tamagui/tamagui/commit/6b0fce3be0e34f4f57744b11358037c8ffddc69a
